### DISCUSSION TITLE
api rules for preventing providing a complete list

### DISCUSSION THREAD

**Discussion by m-2k** [pocketbase/pocketbase#4348] [Q&A]
I get records from collection by some ids:
```js
const filter = ids.map((id) => `id="${id}"`).join(" || ")
return await p.collection("profiles").getFullList({ filter })
```
but for this I need to provide access for `List/Search rule`.
I don't want anyone to be able to get all the records from the table at once without hassle. I want the client to receive records only if he knows their identifiers.
My ugly solution now (List/Search rule):
```dart
@request.query.filter:each ~ "id" && @request.query.filter:each !~ "~"
```

It's okay? lol…

**Comment by ganigeorgiev** [comment]
I don't think the above is reliable. I'd recommend creating a custom route that accepts list of ids and retrieve the specific records OR maybe as an easier workaround create a helper View collection (ex. `profileIds`) with only the allowed fields:
```sql
SELECT id, id as profile
FROM profiles
```
And then on the client-side use something like:

```js
const filter = ids.map((id) => `id="${id}"`).join(" || ")
return await p.collection("profileIds").getFullList({ filter, expand: "profile" }).map((item) => item.expand?.profile).filter(Boolean)
```

**Reply by ganigeorgiev** [reply]
_More advanced field controls like specifying whether the field is "filterable" will be considered during the planned refactoring in the near future._

**Reply by m-2k** [reply]
my code and rules works fine.
I don’t quite understand why I need a View
creating a custom route is easy, but I don't want to

I want the client to receive data only when it **knows** the record's ID.
and prevent calls returning a full list of user profiles:
```js
p.collection("profiles").getFullList() // this will make kittens cry
```

**Comment by laszlo1337** [comment]
> I get records from collection by some ids:
> 
> ```js
> const filter = ids.map((id) => `id="${id}"`).join(" || ")
> return await p.collection("profiles").getFullList({ filter })
> ```
> 
> but for this I need to provide access for `List/Search rule`. I don't want anyone to be able to get all the records from the table at once without hassle. I want the client to receive records only if he knows their identifiers. My ugly solution now (List/Search rule):
> 
> ```dart
> @request.query.filter:each ~ "id" && @request.query.filter:each !~ "~"
> ```
> 
> It's okay? lol…

hello. I was trying to do something similar with collection where I want to get record by its unique property:
`@request.auth.id != '' && @request.query.filter.property = property`

but that doesn't work. I read this as "only allow authenticated request and respond if filter.property is the same as found property" but it's probably wrong. And I have the same issue - disallow searching/listing thru the collection, only return 1 item if I know it's indexing property. Is your solution working like that? because I kinda don't understand it much

**Reply by ganigeorgiev** [reply]
@laszlo1337 `filter` is a query parameter and its value is a plain string, aka. there is no separate treatment or parsing for the property value.

It is not clear what you are trying to do but note that collection fields can be targeted directly, aka. `yourField = 123`.

If you need further help, please don't comment on old unrelated discussions and instead open a new one with clear description of your use case, collections structure and the desired behavior.
